client/src/pages/heroes/radar.tsx

```diff
@@ -0,0 +1,123 @@
+import { PageLayout } from 'components/PageLayout';
+import withSession, { Session } from 'components/withSession';
+import { useCallback, useEffect, useState } from 'react';
+import { GratitudesApi, HeroesRadarDto } from 'api';
+import HeroesRadarTable from 'components/Heroes/HeroesRadarTable';
+import { Form, Select, Button, Checkbox } from 'antd';
+import { useAsync } from 'react-use';
+import { CoursesService } from 'services/courses';
+import { Course } from 'services/models';
+import { onlyDefined } from 'utils/onlyDefined';
+import { IPaginationInfo } from 'common/types/pagination';
+
+type Props = {
+  session: Session;
+};
+
+export type HeroesRadarFormProps = {
+  courseId?: number;
+  notActivist?: boolean;
+};
+
+export type GetHeroesProps = HeroesRadarFormProps & Partial<IPaginationInfo>;
+
+export type LayoutType = Parameters<typeof Form>[0]['layout'];
+
+const initialPage = 1;
+const initialPageSize = 20;
+const initialQueryParams = { current: initialPage, pageSize: initialPageSize };
+
+function Page(props: Props) {
+  const [loading, setLoading] = useState(false);
+  const [heroes, setHeroes] = useState<HeroesRadarDto>({
+    content: [],
+    pagination: { current: initialPage, pageSize: initialPageSize, itemCount: 0, total: 0, totalPages: 0 },
+  });
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [form] = Form.useForm();
+  const [formData, setFormData] = useState<HeroesRadarFormProps>(form.getFieldsValue());
+  const [formLayout, setFormLayout] = useState<LayoutType>('inline');
+  const gratitudeApi = new GratitudesApi();
+
+  const getHeroes = async ({
+    current = initialPage,
+    pageSize = initialPageSize,
+    courseId,
+    notActivist,
+  }: GetHeroesProps) => {
+    try {
+      setLoading(true);
+      const { data } = await gratitudeApi.getHeroesRadar(current, pageSize, courseId, notActivist);
+      setHeroes(data);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getHeroes(initialQueryParams);
+  }, []);
+
+  useAsync(async () => {
+    const [courses] = await Promise.all([new CoursesService().getCourses()]);
+    setCourses(courses);
+  }, []);
+
+  const makeRequest = useCallback(
+    async (data: HeroesRadarFormProps) => {
+      await getHeroes(data);
+    },
+    [heroes],
+  );
+
+  const handleSubmit = useCallback(
+    async (formData: HeroesRadarFormProps) => {
+      const data = onlyDefined(formData);
+      setFormData(data);
+      await makeRequest(data);
+    },
+    [heroes],
+  );
+
+  const onClear = useCallback(async () => {
+    form.resetFields();
+    setFormData(form.getFieldsValue());
+    await getHeroes(initialQueryParams);
+  }, [heroes]);
```

Comment:
not used, but in deps

Reply:
Could you please rephrase? I didn't get what you were trying to say or what specific part of code this comment is related too

Reply:
Artsiom meant that you don't need to include "heroes" in the dependency array. For more details, you can refer to the React documentation.
https://react.dev/reference/react/useCallback#reference

Reply:
> Artsiom meant that you don't need to include "heroes" in the dependency array. For more details, you can refer to the React documentation. https://react.dev/reference/react/useCallback#reference

Got it, thank you

Reply:
Fixed